src/AWS.jl

```diff
@@ -588,6 +586,9 @@ function (service::RestXMLService)(
     end
 
     request.url = generate_service_url(aws_config, request.service, request.resource)
+    if request.service == "s3"
+        request.url = _clean_s3_uri(request.url)
```

Comment:
Note that `request_uri` would be something like `/my-bucket/my=file`, while the full URL would be something like `http://s3.us-east-1.amazonaws.com/my-bucket/my=file/?prefix=foobar` or something like that. We only want to scrub the URI otherwise we would be cleaning `prefix=foobar` which will break.

You would want this instead, also might be worthwhile to turn it into a `!` function.

```suggestion
        request_uri = _clean_s3_uri(request_uri)
```

Then line should also be moved above line 577 (in this commit) where the `request.resource` is set.

Reply:
Thanks! 

> might be worthwhile to turn it into a `!` function

I don't think this is possible, since a `String` is not mutable

Reply:
Oh right, :4head: